control-plane/connect-inject/endpoints_controller.go

```diff
@@ -639,15 +658,28 @@ func (r *EndpointsController) createGatewayRegistrations(pod corev1.Pod, service
 				Address: wanAddr,
 				Port:    wanPort,
 			},
-		},
+		}
+	case TerminatingGateway:
+		service.Kind = api.ServiceKindTerminatingGateway
+		service.Service = TerminatingGateway
```

Comment:
Sorry I just spotted this. This has to be the name of the service that maps to this endpoint. It will need to be different for each terminating gateway.